.github/workflows/run_tests.yml

```diff
@@ -50,18 +43,42 @@ jobs:
           - 'fedora:36'
           # EOL May 2024 https://www.centos.org/centos-stream/
           - 'quay.io/centos/centos:stream8'
-        cc: ['gcc', 'clang']
+        cc: ['gcc']
         buildtype: ['debug', 'release']
-        exclude:
-          # https://github.com/shadow/shadow/issues/1741
-          - container: 'quay.io/centos/centos:stream8'
-            cc: clang
-          # https://github.com/shadow/shadow/issues/1741
-          - container: 'ubuntu:21.10'
-            cc: clang
-          # https://github.com/shadow/shadow/issues/1741
-          - container: 'fedora:35'
-            cc: clang
+        include:
+          # Run some tests on the newest-available base vm, and hence
+          # newest-available kernel. https://github.com/actions/virtual-environments
+          - vm: 'ubuntu-22.04'
+            container: 'ubuntu:22.04'
+            cc: 'gcc'
+            buildtype: 'release'
+          - vm: 'ubuntu-22.04'
+            container: 'ubuntu:22.04'
+            cc: 'gcc'
+            buildtype: 'debug'
+
+          # Run some tests on the newest-available clang.  Testing clang on
+          # *every* platform is a bit overkill, but testing on at least one
+          # gives decent "bang for the buck" of testing compatibility with
+          # clang's most recent diagnostics and optimizations.
```

Comment:
I think we should also run a test on a non-debian distro (maybe fedora), and on a distro with an older version of clang. Some distros have different clang warnings enabled by default, and since we use `-Wall` in the CI, we'd catch those.

Reply:
> Some distros have different clang warnings enabled by default, and since we use -Wall

Doesn't `-Wall` override the defaults?

In general I think newer versions of clang will catch the most problems; testing on older versions have pretty diminishing returns.

Going to go ahead and merge this now while I'm in a meeting with @robgjansen so we can coordinate changing the required actions, but happy to revisit